packages/core/src/types.ts

```diff
@@ -337,7 +337,9 @@ export type ExtractRematchDispatcherFromReducer<
 > = TReducer extends () => any
 	? RematchDispatcher
 	: TReducer extends (state: TState) => TState
-	? RematchDispatcher
+	? Parameters<TReducer> extends [TState]
+		? RematchDispatcher
+		: RematchDispatcher<Parameters<TReducer>[1] | undefined>
```

Comment:
How about just `RematchDispatcher<Parameters<TReducer>[1]>` cause `Parameters<TReducer>[1]` include `undefined` in this branch.

Reply:
nice catch, PR updated